### PR TITLE
feat: Price Daemon

### PR DESCRIPTION
**Pull request by davidterpay**

### CODE REVIEW

File: oracle/config/app.go [hunk 1]
```diff
@@ -65,6 +85,12 @@ type AppConfig struct {
 
 	// MetricsEnabled is a flag that determines whether oracle metrics are enabled.
 	MetricsEnabled bool `mapstructure:"metrics_enabled" toml:"metrics_enabled"`
+
+	// MaxAge is the maximum age of the latest price response before it is considered stale.
+	MaxAge time.Duration `mapstructure:"max_age" toml:"max_age"`
```
Comment (Eric-Warehime) [inline oracle/config/app.go]: `oracle.max_age` feels pretty vague in terms of describing what this does. How about `oracle.price_ttl` instead or something along those lines?

File: oracle/config/app.go [hunk 2]
```diff
@@ -74,11 +100,23 @@ func (c *AppConfig) ValidateBasic() error {
...
+	if c.MaxAge <= 0 || c.MaxAge > MaxMaxAge {
+		return fmt.Errorf("poorly formatted app.toml (oracle subsection): oracle max age must be between 0 and %s", MaxMaxAge)
+	}
+
+	if c.Interval <= 0 || c.Interval > MaxInterval {
+		return fmt.Errorf("poorly formatted app.toml (oracle subsection): oracle interval must be between 0 and %s", MaxInterval)
+	}
+
+	if c.Interval >= c.MaxAge {
+		return fmt.Errorf("poorly formatted app.toml (oracle subsection): oracle interval must be strictly less than max age")
```
Comment (Eric-Warehime) [inline oracle/config/app.go]: I think it's good to have these validations, but we should be setting defaults for people. Nobody should really be touching these on their own.

**Review by Eric-Warehime — COMMENTED** [review]
LGTM, but we should set default values and make this backwards compatible.

**Review by Eric-Warehime — APPROVED** [review]

File: .github/workflows/e2e.yml [hunk 3]
```diff
@@ -12,7 +12,7 @@ jobs:
       - uses: actions/checkout@v4
       - uses: actions/setup-go@v5
         with:
-          go-version: 1.22.5
+          go-version: 1.22.3
```
Comment (aljo242) [inline .github/workflows/e2e.yml]: why this change?

File: oracle/config/app.go [hunk 4]
```diff
@@ -120,27 +150,53 @@ func ReadConfigFromAppOpts(opts servertypes.AppOptions) (AppConfig, error) {
 		if cfg.Enabled, err = cast.ToBoolE(v); err != nil {
 			return cfg, err
 		}
+	} else {
+		cfg.Enabled = DefaultOracleEnabled // set to default
```
Comment (aljo242) [inline oracle/config/app.go]: can we have all of these values set to default before we overwrite w flags

File: service/clients/oracle/daemon.go [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+			time.Since(ts).String(),
+		)
+	}
+
+	return latest, nil
+}
+
+// Stop stops the price daemon.
+func (d *PriceDaemon) Stop() error {
+	d.doneCh <- struct{}{}
```
Comment (aljo242) [inline service/clients/oracle/daemon.go]: should close this channel too

File: service/clients/oracle/daemon.go [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			d.logger.Info("stopping price daemon from context")
+			return ctx.Err()
+		case <-d.doneCh:
+			d.logger.Info("price daemon stopped")
+			close(d.doneCh)
```
Comment (aljo242) [inline service/clients/oracle/daemon.go]: I would close the channel in the `Stop()` function.  This sends a final signal on the channel so this case will still be met

File: .github/workflows/e2e.yml (hunk 3 above)
Comment (davidterpay) [reply .github/workflows/e2e.yml]: thought this was related to to the docker issue, i can revert back to .5.

File: service/clients/oracle/daemon.go (hunk 5 above)
Comment (davidterpay) [reply service/clients/oracle/daemon.go]: the channel gets closed in `Stop()`. can close it here as well but issue is that this would block if start isn't executed first.
